Feature/angular 8 support

christophercr opened the pull request:
ISSUES CLOSED: #31 

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/ngx-form-errors/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [X] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #31 

## What is the new behavior?
- `ngx-form-errors` officially supports Angular 8 now.
- the demo is now implemented in the 2 latest versions of Angular: 7 and 8.
- the installation, linting and unit testing of the demo apps for the 2 latest angular versions are included in the `travis.yml` file.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information
@SuperITMan @nicanac you can see that the second commit of this PR adds two subfolders under `/demo` containing the demo app in the latest Angular version (8.x) and the previous one (7.x).

I suggest to keep these 2 versions of the demo app and maintain them with every new major version of Angular. For example, once Angular 9 is out, we keep the demo app in Angular 8 and 9, and so on.

Cause maybe with a new major version of Angular, it might not be possible to keep support for an old version. So at least we keep our demo with the 2 latest major versions so we are aligned with the [Angular support schedule](https://angular.io/guide/releases#support-policy-and-schedule).

christophercr commented:
Well, it seems we need to change the `node` versions we use in Travis because the [node 8 build](https://travis-ci.org/NationalBankBelgium/ngx-form-errors/jobs/612375350) of this PR is failing due to:

```
You are running version v8.16.2 of Node.js, which is not supported by Angular CLI 8.0+.
The official Node.js version that is supported is 10.9 or greater.
```

SuperITMan commented:
The maintenance of Node 8 ends in January 2020 and in NBB we are now using Node 10 for our projects.
I think that we can drop the support of Node 8 for ngx-form-errors and for Stark and replace it by Node 12. 
What do you think @christophercr @nicanac? 😊

nicanac commented:
> The maintenance of Node 8 ends in January 2020 and in NBB we are now using Node 10 for our projects.
> I think that we can drop the support of Node 8 for ngx-form-errors and for Stark and replace it by Node 12.
> What do you think @christophercr @nicanac? 😊

yes, I fully agree. We don't need the version 8 anymore.

christophercr commented:
Ok, so I'll replace node 8 by 12 😉

christophercr commented:
Ok, node version changed and Travis builds passing! 👍

SuperITMan commented on demo-app/ng8/src/app/pages/reactive-forms-example/reactive-forms-example.component.ts:
```diff
@@ -0,0 +1,92 @@
+		this.formGroup = this.formBuilder.group({
+			username: [undefined, Validators.required],
+			matchingPasswords: new FormGroup(
+				{
```
The indentation is not the same in both demo-apps (ng7, ng8).
Could you please fix it and to have "2 spaces" everywhere ? 😊

christophercr replied:
Sorry, I don't get it. What "2 spaces"?

I've ran Prettier on the whole project and the 2 demo-apps have exactly the code formatting...

SuperITMan replied:
@christophercr my apologies...
I did the review with GitHub and I thought that the indentation was incorrect. But it's not. So please forget this comment 😊

SuperITMan commented on demo-app/ng8/src/polyfills.ts:
```diff
@@ -0,0 +1,74 @@
+ * APPLICATION IMPORTS
+ */
+// FixMe: Remove when NBB upgrades chrome browser (https://caniuse.com/#search=grid)
+import "./polyfills/css-grid-layout-polyfill";
```
Should we remove this import, and the file "css-grid-layout-polyfill", since we are using Chrome >v74 at NBB ? 😊 
Or should we keep it for another reason?

christophercr replied:
Ah true, we should remove it. I'll remove it and test it also in IE 😉

christophercr replied:
I've tested and indeed, we don't need the polyfill for Chrome, however it is necessary for IE but... the problem in IE with this polyfill is that it causes some flickering which is quite annoying.

So I think we should just remove the polyfill and let IE display the grid the best it can... which doesn't look too bad in my opinion, you know... it's IE 😕

christophercr commented:
@SuperITMan @nicanac I've updated this PR with a couple of changes:

- removed the CSS grid polyfill as suggested by @SuperITMan because it is no longer needed with the latest versions of Chrome (including the one used at the NBB).
- disabled the `host check` from `webpack-dev-server` as mentioned [here](https://github.com/webpack/webpack-dev-server/issues/1604#issuecomment-453137871) due to an error being logged constantly in IE console.
